### PR TITLE
Add lint aspect for Ruff

### PR DESCRIPTION
**Pull request by tokongs**
This project looks interesting! 

I thought I'd try it out by creating a lint aspect for Ruff. And with that done, I might as well submit a PR for it too. 

This adds support for the Ruff Python linter. I was unsure how you'd like to structure the binary download in the example.

---
### Type of change

- New feature or functionality (change which adds functionality)

**For changes visible to end-users**
- Relevant documentation has been updated

### Test plan
- Covered by existing test cases

### CODE REVIEW

File: docs/BUILD.bazel [hunk 1]
```diff
@@ -17,4 +17,9 @@ stardoc_with_diff_test(
     bzl_library_target = "//lint:flake8",
 )
 
+stardoc_with_diff_test(
+    name = "ruff",
+    bzl_library_target = "//lint:ruff",
```
Comment (alexeagle) [inline docs/BUILD.bazel:22]: the resulting `ruff.md` file should also be git add'ed

**Review by alexeagle — COMMENTED** [review]
Awesome thanks! Could you rebase on main?

File: docs/BUILD.bazel (hunk 1 above)
Comment (alexeagle) [reply docs/BUILD.bazel:22]: and also add ruff to the /README.md

File: example/BUILD.bazel [hunk 2]
```diff
@@ -45,3 +47,15 @@ java_binary(
     main_class = "net.sourceforge.pmd.PMD",
     runtime_deps = ["@net_sourceforge_pmd"],
 )
+
+# We can test that it works with:
+# bazel run :ruff -- help
+native_binary(
+    name = "ruff",
+    src = select({
+        "@platforms//os:osx": "@ruff_bin_darwin_arm64//:ruff",
```
Comment (alexeagle) [inline example/BUILD.bazel:56]: We'll need more platforms like darwin x86 - we should mirror all the ones the ruff project publishes. Okay for a follow-up though. We should have a script that does the mirroring (and can automate it sending us PRs)

**Review by alexeagle — APPROVED** [review]
Thanks!

**Comment by alexeagle** [comment]
Oh, can you sign our CLA also? We need legal permission to use your work.

**Comment by tokongs** [comment]
@alexeagle I'm sadly not quite sure I'll be able to sign the CLA with the current License the project has, as we will not be able to use it.

**Comment by tokongs** [comment]
I'll close this for now, but I can re-open it in the future if the license changes. Thanks for the rapid feedback and sorry for not seeing the license earlier!

**Comment by alexeagle** [comment]
This repo is now licensed Apache 2.0, largely because I think we'll only get critical mass of linter integrations with contributions like this one :)

**Comment by tokongs** [comment]
Nice! I've signed the CLA now :)

**Pull request merged**